src/compact.rs

```diff
@@ -63,7 +64,7 @@ impl<'a, T: 'a + Input> Input for PrefixInput<'a, T> {
 
 	fn read(&mut self, buffer: &mut [u8]) -> Result<(), Error> {
 		match self.prefix.take() {
```

Comment:
The correct solution would be that the check `if buffer.is_empty()` is moved before the match.